mkosi/distributions/serpentos.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import re
+import shutil
+import textwrap
+from collections.abc import Sequence
+
+from mkosi.config import Architecture
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer.moss import invoke_moss
+from mkosi.log import die
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "Serpent OS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "btrfs"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.moss
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "rolling"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.serpentos
+
+    @classmethod
+    def _move_kernel_and_initrd(cls, context: Context) -> None:
+        regexp = re.compile("com\.serpentos\.([a-z][a-z0-9]+)\.(.*)")
+
+        # SerpentOS puts its kernels and initrds into /usr/lib/kernel
+        kernel_dir = context.root / "usr/lib/kernel"
+        for k in kernel_dir.glob("com.serpentos.*"):
+            match = regexp.fullmatch(str(k.name))
+            if not match:
+                continue
+
+            kkind = match.group(1)
+            kver = match.group(2)
+
+            modules_dir = (context.root / f"usr/lib/modules/{kver}.{kkind}")
+            if modules_dir.is_dir():
+                shutil.move(k, (modules_dir / f"vmlinuz-{kver}.{kkind}"))
+                shutil.move((kernel_dir / f"initrd-com.serpentos.{kkind}.{kver}"),
+                            (modules_dir / f"initrd-{kver}.{kkind}"))
+
+                # Move the config, etc. while at it anyway:
+                for f in kernel_dir.glob("*{kver}.{kkind}"):
+                    shutil.move(f, (modules_dir / f.name))
+
+                # Remove now dead symlinks to current kernel/initrd
+                for f in kernel_dir.glob("current.*.kernel"):
+                    os.remove(f)
+                for f in kernel_dir.glob("current.*.initrd"):
+                    os.remove(f)
+
+    @classmethod
+    def setup(cls, context: Context) -> None:
+        # needed to normalize modification time of /boot later:
+        (context.root / "boot").mkdir(exist_ok=True)
+
+        repo_dir = (context.root / "etc/moss/repo.d")
+        if not repo_dir.exists():
+            repo_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
No point in checking if this exists if we're going to create it unconditionally. You should check instead if `volatime.yaml` already exists and only write it if it doesn't already exist.

Reply:
This makes it harder to extend the "volatile" repo with other repos when that becomes possible. Please name the file `mkosi.yml` so that users can override the default repositories by simply creating an empty file `mkosi.yml` in their package manager tree.